### PR TITLE
Expose RDLog to SWIG wrappers

### PR DESCRIPTION
**Pull request by ptosco**
The logger could not be initialized from Java/C#, so no logging was available.

### CODE REVIEW

**Comment by greglandrum** [comment]
@ptosco : there's no testing code here. That's fine for something with the logger, but have you verified that this actually works?

**Comment by ptosco** [comment]
@greglandrum I have, but you are right that I forgot to add a test - I will add it later tonight, hold off a bit before merging.

**Comment by ptosco** [comment]
@greglandrum I have added C++, Java and C# tests. To make the logs accessible and more useful from Java and C# I have added a `SetTee()` overload that takes a filename as parameter, since there is no easy way to access a C++ stream from Java and C#. I also had to switch `std:;shared_ptr` to `boost::shared_ptr` as SWIG allows wrapping either, but not both at the same time (source: https://www.swig.org/Doc3.0/Library.html#Library_std_shared_ptr); I don't think this should not be a problem.
This is now ready for review.

**Comment by ptosco** [comment]
@greglandrum No, it is not - it is segfaulting. Let me figure out what's up.

**Comment by ptosco** [comment]
@greglandrum OK, this is now _really_ ready for review.

**Comment by ptosco** [comment]
@greglandrum Sorry to pester but it looks like this PR is not getting much love :-)

**Comment by greglandrum** [comment]
@ptosco I have actually spent some time on it, because I really, really don't want to switch the loggers back to `boost::shared_ptrs`. I think we should be able to handle this in another way by just exposing naked  pointers to SWIG, but I haven't quite finished making that work yet.

**Comment by ptosco** [comment]
@greglandrum Ah, sorry about that, I didn’t know! I can also take care of that, it shouldn’t be a major issue.

**Comment by ptosco** [comment]
@greglandrum I have resurrected this old PR in a way which looks good to me, since the SWIG objects do not own the `rdLogger` pointer. What do you think?

**Comment by ptosco** [comment]
@greglandrum If you have a second to look at this one it would be great, so if you think it's fine it can be merged. Thanks!

File: Code/RDGeneral/RDLog.h
Comment (greglandrum) [inline Code/RDGeneral/RDLog.h:36]: please rename this to something `dp_teeHelperStream` so that it's consistent with our usual naming scheme

**Review by greglandrum — CHANGES_REQUESTED** [review]
Looks good other than the naming of the local

File: Code/RDGeneral/RDLog.h
Comment (ptosco) [reply Code/RDGeneral/RDLog.h]: Done.

**Review by greglandrum — APPROVED** [review]
LGTM

**Pull request merged**
